### PR TITLE
Fix #799 by forcing no setter for read-only fields

### LINKED ISSUE #799: Wrong behavior when using var on readonly fields in Kotlin
Related to #680

When using the modifier ```var``` instead of ```val``` on implicit or explicit read-only fields, no setter methods get generated if you use abstract classes resulting in an error that they are missing. However, if you use interfaces they will get generated. In both cases Kapt generates method stubs for getters **and** setters.

Requery should result in an error if ```var``` instead of ```val``` is used on read-only fields. This should also apply for interfaces. One just needs to test whether a specific setter is present or not.

Correct behavior would be:
1. test whether field should be read-only
1.1 if it is check for present setters and throw an error *or* 
1.2 overwrite and proceed as normal

Example:
```kotlin
@get:Key
@get:Generated
var id: Int
```
results in (by kapt)
```java
@io.requery.Generated()
@io.requery.Key()
public abstract int getId();

public abstract void setId(int p0);
```
but ```setId()``` wont get overriden because ```id``` is generated and therefore read-only.

### PR DESCRIPTION
**Pull request by lehmanju**
Implements behavior as described in #799. Corresponding test files have been adjusted.

(pls ignore wrong branch name ;)

### CODE REVIEW

**Comment by lehmanju** [comment]
Now it basically just checks interfaces for present setter methods.

**Comment by lehmanju** [comment]
Also works with java classes and interfaces

File: requery-processor/src/main/java/io/requery/processor/AttributeMember.java [hunk 1]
```diff
@@ -159,6 +159,8 @@
         isEmbedded = annotationOf(Embedded.class).isPresent() ||
             annotationOf(javax.persistence.Embedded.class).isPresent();
         indexNames.forEach(name -> checkReserved(name, validator));
+        if (isReadOnly)
```
Comment (npurushe) [inline requery-processor/src/main/java/io/requery/processor/AttributeMember.java]: Please add braces around one-line if statements

**Comment by lehmanju** [comment]
Done

**Pull request merged**